### PR TITLE
PR feedback

### PR DESCRIPTION
**Pull request by orangejenny**
https://github.com/dimagi/Vellum/pull/673/files/18ad819633187312792b90643426e31a06c55d6a#r67002157

@millerdev

### CODE REVIEW

File: src/datasources.js [hunk 1]
```diff
@@ -128,11 +134,13 @@ define([
                     success: finish,
                     error: function (jqXHR, errorType, exc) {
                         finish([]);
-                        if (errorCallback) {
+                        _.each(errorCallbacks, function(callback) {
                             errorCallback(jqXHR, errorType, exc);
```
Comment (millerdev) [inline src/datasources.js]: I think this should be `callback(jqXHR, errorType, exc);`

Also, might want to check if `callback` is not undefined before calling it.

File: src/datasources.js [hunk 2]
```diff
@@ -128,11 +134,13 @@ define([
...
-                        } else {
+                        });
+                        if (!errorCallbacks) {
```
Comment (millerdev) [inline src/datasources.js]: I think this should say `if (!_.some(errorCallbacks))`

File: src/datasources.js (hunk 1 above)
Comment (orangejenny) [inline src/datasources.js]: oops, thanks. fixed.

File: src/datasources.js (hunk 2 above)
Comment (orangejenny) [inline src/datasources.js]: updated so that the callbacks arrays are always arrays and never null

**Pull request merged**